components/RecentAuctions/RecentAuctions.tsx

```diff
@@ -81,6 +116,50 @@ function RecentAuctions(props: Props) {
         loadRecentAuctions()
     }
 
+    function onAfterLogin() {
+        api.getPremiumProducts().then(products => {
+            setIsLoggedIn(true)
+            let highestPremium = getPremiumType(getHighestPriorityPremiumProduct(products))
+            premiumType = highestPremium
+            setPremiumType(() => {
+                setAllElementsLoaded(() => {
+                    if (highestPremium !== null) {
+                        loadRecentAuctions()
+                    }
+                    return false
+                })
+                return highestPremium
+            })
+        })
+    }
+
+    function getMoreRecentAuctionsElement() {
+        if (!isLoggedIn || !premiumType) {
+            return (
+                <p>
+                    You can see more auctions with{' '}
+                    <Link href={'/premium'} style={{ marginBottom: '15px' }}>
+                        <a>Premium:</a>
+                    </Link>
+                    <GoogleSignIn onAfterLogin={onAfterLogin} />
+                </p>
+            )
+        }
+        if (premiumType.priority === PREMIUM_RANK.STARTER) {
+            return (
+                <p>
+                    You currently use Starter Premium. You can see up to 120 active auctions with
+                    <Link href={'/premium'}>
+                        <a>Premium:</a>
+                    </Link>
+                    <div style={{ marginTop: '15px' }}>
+                        <GoogleSignIn onAfterLogin={onAfterLogin} />
+                    </div>
+                </p>
+            )
+        }
```

Comment:
This is fully redundant with the active auctions function. Please make it less redundant. (you also didn't change `active auctions` in line 151)